### DISCUSSION TITLE
Supplied javaHome must be a valid directory. You supplied: /Applications/Android         Studio.app/Contents/jre/jdk/Contents/Home

### DISCUSSION THREAD

**Discussion by marcarausch** [ionic-team/capacitor#5171] [Q&A] [answered]
I am trying to run npx cap run android and I am getting this error:

Supplied javaHome must be a valid directory. You supplied: /Applications/Android
        Studio.app/Contents/jre/jdk/Contents/Home

I have this entry in my .bash_profile
export JAVA_HOME=$(/usr/libexec/java_home)

if I run echo $JAVA_HOME in my terminal this is location of the jdk
/Library/Java/JavaVirtualMachines/jdk1.8.0_271.jdk/Contents/Home

Anyone have any suggestions on this issue?

I have this setting in my android studio Project settings:

![image](https://user-images.githubusercontent.com/14215422/138832856-c710da0c-01f4-45f8-8423-856331a2031c.png)

I am running Android Studio Artic Fox 2020.3.1

**Comment by jcesarmobile** [comment]
What's your OS version?
I've read that using `/usr/libexec/java_home` have some problems on Big Sur, I just set the JAVA_HOME to the java path directly.

Also, in Catalina Apple moved to zsh shell, while you could still use the old one, not sure if that's still possible on Big Sur, so try using the `.zshrc` file instead of the `.bash_profile`

**Comment by lindsayevans** [comment, marked as answer]
@marcarausch I just started getting the same error after upgrading Android Studio - solution for me was to remove the existing Gradle home directory:
`rm -rf ~/.gradle`

**Reply by marcarausch** [reply]
Thank you so much, this worked for me. Much appreciated !

**Reply by GertBoer** [reply]
Had the same problem and this worked also for me! Thanks!
